internal/update/check.go

```
@@ -0,0 +1,158 @@
+	return c, true
+}
+
+// writeCache atomically writes the cache file with 0700 dir + 0600 file
+// perms — same pattern as config.Save so permissions stay consistent across
+// ana's on-disk state.
+func writeCache(path string, c cacheFile) error {
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, 0o700); err != nil {
+		return fmt.Errorf("update: mkdir %s: %w", dir, err)
+	}
+	data, _ := json.Marshal(c)
```

golangci-lint (2.11.4)

[error] Error return value of `encoding/json.Marshal` is not checked: unsafe type `time.Time` found

(errchkjson)